data_analysis/model_ad/notebooks/MG-190_Create_ui_info_source_file.rmd

```diff
@@ -11,28 +11,132 @@ ADDING NEW PAGES
-2. That's it, unless there is some custom logic related to that new menu option.
+4. Define a vector of values shared by the dynamic columns on the new page (typically all but `name` and possibly `tooltip`); see Disease Correlation for an example that adds column-type-specific tooltips to the dynamic columns
+5. Define a vector of `name` values for each dynamic column on the new page; see Gene Expression for an example that uses a multiple vectors of names to generate unique sets of dynamic columns based on dropdown selection
+6. Define a vector of filterset display names and filterset display values
+7. Create a `build_<page>_objects` function that generates the required columns and filters data
+8. Update the `Combine and Output` section to invoke the new build function
 
 
+## Library setup
+Install required packages:
+- synapser (also requires a Synapse account with an auth token)
+- jsonlite
+- tidyverse
 ```{r}
+if (!require("synapser", quietly = TRUE)) {
+  install.packages("synapser", repos = c("http://ran.synapse.org",
+                                         "https://cloud.r-project.org"))
+}
+if (!require("jsonlite", quietly = TRUE)) {
+  install.packages("jsonlite")
+}
+if (!require("tidyverse", quietly = TRUE)) {
+  install.packages("tidyverse")
+}
+
+
+library(synapser)
 library(jsonlite)
+library(tidyverse)
+```
+
+## Synapse helper functions
+```{r}
+# MODEL-AD 2.0 harmonized source file directory
+# ADP Backend > Files > 2.ConsortiumStudies > MODEL-AD Data Explorer > Model AD Explorer 2.0 Source Files
+synapse_folder <- 'syn51498054'
+
+# Vector to store all downloaded synIds for Synapse provenance (populated via download_file function)
+data_sources <- character()
+
+# Download a file from Synapse by synId and read it into a dataframe
+download_file <- function(synId, df_name, type) {
+
+  synLogin()
+
+  # Capture all downloaded synIds for provenance
+  data_sources <<- c(data_sources, synId)
+
+  input_dir <- file.path("./", "input")
+  id_parts <- strsplit(synId, '.', fixed = TRUE)
+  id_value <- id_parts[[1]][1]
+  version_value <- id_parts[[1]][2]
+
+  if (is.na(version_value)) {
+    version_value <- NULL
+  }
+
+  cat(paste0("Downloading ", synId, "..."))
+
+  file <- synGet(id_value, version = version_value, downloadLocation = input_dir, ifcollision='overwrite.local')
+  path <- file$path
+
+  cat("\nDownload on:  ", date())
+  cat("\n", synId, "\n- Modified on ", file$properties$modifiedOn, "\n- Version ", file$properties$versionNumber)
+  cat("\npath: ", file$path, "\n")
+
+  if (type == 'f') {
+    data <- read_feather(path)
+  } else {
+    if (type == 'json') { data <- fromJSON(path) }
+    if (type == 'csv' || type == 'table') { data <- read.csv(path) }
+    if (type == 'tsv') { data <- read.csv(path, sep = "\t") }
+    if (type == 'rda') { data <- load(path)}
+    if (type == 'rds') {data <- readRDS(path)}
+    df_name <- as.data.frame(data)
+  }
+  df_name
```

Comment:
This doesn't work if `type == 'f'`, that variable is called "data". Or is `df_name <- as.data.frame(data)` supposed to apply to that one too? That's why I was a little confused by this. Why is feather its own thing while everything else is under a single `else` statement with a different output variable?

Reply:
Haha, nice. Ok thanks for fixing!